[BUG] v 7.3.0 when publishing to private rep You need to authorize this machine using 'npm adduser'

### Current Behavior:
Can not publish(This command requires you to be logged in) but I  adduser  successfully 
### Expected Behavior:
 - it work with npm v 6

### Steps To Reproduce:
-  npm adduser --registry=https://zelenjak.arnes.si/nexus/repository/ArnesNpm/
- ok login and in my .npmrc I get
```
registry=https://zelenjak.arnes.si/nexus/repository/ArnesGroupNPM/
//zelenjak.arnes.si/nexus/repository/ArnesNpm/:_authToken=NpmToken.XXXXXXXXX
```
- in my package I have:
```
,
  "publishConfig": {
    "registry": "https://zelenjak.arnes.si/nexus/repository/ArnesNpm/"
  },
```
- when running npm publish I got error:
```
npm ERR! code ENEEDAUTH
npm ERR! need auth This command requires you to be logged in.
npm ERR! need auth You need to authorize this machine using `npm adduser`
```

### Environments:

- Node: 12.6.0
- npm: 7.3.0
- OS: win32 x64

--- a/lib/publish.js
+++ b/lib/publish.js
@@ -35,22 +35,7 @@ const publish = async args => {
   log.verbose('publish', args)
 
   const opts = { ...npm.flatOptions }
-  const { json, defaultTag, registry } = opts
-
-  if (!registry) {
-    throw Object.assign(new Error('No registry specified.'), {
-      code: 'ENOREGISTRY',
-    })
-  }
-
-  if (!opts.dryRun) {
-    const creds = npm.config.getCredentialsByURI(registry)
-    if (!creds.token && !creds.username) {
-      throw Object.assign(new Error('This command requires you to be logged in.'), {
-        code: 'ENEEDAUTH',
-      })
-    }
-  }
+  const { json, defaultTag } = opts
 
   if (semver.validRange(defaultTag))
     throw new Error('Tag name must not be a valid SemVer range: ' + defaultTag.trim())
@@ -90,6 +75,22 @@ const publish_ = async (arg, opts) => {
 
   if (manifest.publishConfig)
     Object.assign(opts, publishConfigToOpts(manifest.publishConfig))
+
+  const { registry } = opts
+  if (!registry) {
+    throw Object.assign(new Error('No registry specified.'), {
+      code: 'ENOREGISTRY',
+    })
+  }
+
+  if (!dryRun) {
+    const creds = npm.config.getCredentialsByURI(registry)
+    if (!creds.token && !creds.username) {
+      throw Object.assign(new Error('This command requires you to be logged in.'), {
+        code: 'ENEEDAUTH',
+      })
+    }
+  }
 
   // only run scripts for directory type publishes
   if (spec.type === 'directory') {

--- a/test/lib/publish.js
+++ b/test/lib/publish.js
@@ -369,3 +369,44 @@ t.test('throw if not logged in', async t => {
     }, 'throws when not logged in')
   })
 })
+
+t.test('read registry only from publishConfig', t => {
+  t.plan(3)
+
+  const publishConfig = { registry: 'https://some.registry' }
+  const testDir = t.testdir({
+    'package.json': JSON.stringify({
+      name: 'my-cool-pkg',
+      version: '1.0.0',
+      publishConfig,
+    }, null, 2),
+  })
+
+  const publish = requireInject('../../lib/publish.js', {
+    '../../lib/npm.js': {
+      flatOptions: {
+        json: false,
+      },
+      config,
+    },
+    '../../lib/utils/tar.js': {
+      getContents: () => ({
+        id: 'someid',
+      }),
+      logTar: () => {},
+    },
+    '../../lib/utils/output.js': () => {},
+    libnpmpublish: {
+      publish: (manifest, tarData, opts) => {
+        t.match(manifest, { name: 'my-cool-pkg', version: '1.0.0' }, 'gets manifest')
+        t.same(opts.registry, publishConfig.registry, 'publishConfig is passed through')
+      },
+    },
+  })
+
+  return publish([testDir], (er) => {
+    if (er)
+      throw er
+    t.pass('got to callback')
+  })
+})

--- a/test/lib/publish.js
+++ b/test/lib/publish.js
@@ -329,7 +329,6 @@ t.test('throw if no registry', async t => {
     '../../lib/npm.js': {
       flatOptions: {
         json: false,
-        defaultTag: '0.0.13',
         registry: null,
       },
       config,
@@ -350,7 +349,6 @@ t.test('throw if not logged in', async t => {
     '../../lib/npm.js': {
       flatOptions: {
         json: false,
-        defaultTag: '0.0.13',
         registry: 'https://registry.npmjs.org/',
       },
       config: {

